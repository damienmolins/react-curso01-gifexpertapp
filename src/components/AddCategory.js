import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        console.log('Handle InputChange...');
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        console.log('Handle Submit...');
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <h1>{inputValue}</h1>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}>
            </input>
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}