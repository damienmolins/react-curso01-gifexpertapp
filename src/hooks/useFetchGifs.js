import { useState, useEffect } from "react";
import { getGifs } from "../helpers/GetGifs";

export const useFetchGifs = (category) => {
    const [state, setState] = useState(
        {
            data: [],
            loading: true
        });

    //Se puede hacer tranquilamente sin el setTimeout...
    useEffect(() => {
        getGifs(category)
            .then(imgs => {
                setTimeout(() => {
                    //console.log(imgs);
                    setState({
                        data: imgs, loading: false
                    });
                }, 2000);
            })
    }, [category])

    return state;
}
