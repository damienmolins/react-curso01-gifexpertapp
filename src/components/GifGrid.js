import { GifGridItem } from './GifGridItem';
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';


export const GifGrid = ({ category }) => {

    const { data: imagenes, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {loading && <p className="animate__animated animate__flash">Cargando...</p>}

            <div className="card-grid">
                {
                    imagenes.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>

        </>
    );
}
