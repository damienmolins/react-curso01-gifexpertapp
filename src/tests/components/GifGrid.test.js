import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';
jest.mock('../../hooks/useFetchGifs');

describe('pruebas sobre <GifGrid />', () => {

    const category = 'Dragon Ball';

    test('snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes...', () => {
        const gifs = [
            {
                id: 'ABC',
                url: 'https:cualquierurl.com/algo.jpg',
                title: 'Cualquier cosa'
            }
        ];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });


})
