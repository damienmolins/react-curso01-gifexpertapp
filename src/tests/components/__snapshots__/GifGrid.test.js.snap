// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre <GifGrid /> snapshot 1`] = `
<Fragment>
  <h3>
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
